tests/integration/index/query_with_index_only_filter_test.go

```diff
@@ -0,0 +1,395 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package index
+
+import (
+	"testing"
+
+	testUtils "github.com/sourcenetwork/defradb/tests/integration"
+)
+
+func TestQueryWithIndex_WithNonIndexedFields_ShouldFetchAllOfThem(t *testing.T) {
```

Comment:
I guess your concern mainly about `sendRequestAndExplain` function. Unfortunately go doesn't support named parameters, so we can write like this:
```go
sendRequestAndExplain(requestBody: `...`, expectedResult: []map[string]..., asserter: testUtil.New...)
```
I know this way it's not quite explicit when reading the first time, but this way we can avoid writing like this:
```go
...
testUtils.Request{
	Request: `
		query {
			User(filter: {name: {_eq: "Islam"}}) {
			        name
			        age
      			}
		}`,
	Results: []map[string]any{{
	        "name": "Islam",
	        "age":  uint64(32),
        }},,
},
testUtils.Request{
	Request: `
		query @explain(type: execute) {
			User(filter: {name: {_eq: "Islam"}}) {
			        name
			        age
      			}
		}`,
	Asserter:  testUtils.NewExplainAsserter().WithDocFetches(1).WithFieldFetches(2).WithIndexFetches(1),
},
```
This makes it repetitive and error-prone because one can change one request and forget to change the other.

Do you propose to use an arbitrary struct that will explicitly show the parameters? Or maybe a new test action, or to alter an existing one?